### DISCUSSION TITLE
change baseBranch while having open MR

### DISCUSSION THREAD

**Discussion by alreadyhaveanaccount4** [renovatebot/renovate#20774] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

None

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have an open MR to merge `renovate/update` into `old_branch` (baseBranch = old_branch). 
When changing `baseBranch` to `new_branch` and running Renovate again, I expect renovate to 
 - either update the open MR from "merge into old_branch" to "merge into new_branch"
 - or create a new MR with same Title but merge into new_branch
The retry/rebase-checkbox is checked.

Currently, the open MR is just being rebased with `new_branch` but if I would merge it, it would still get merged into `old_branch`, while it should be merged into `new_branch`.

Do I need to close the MR or delete the renovate-branch?

### What you tried so far.

I changed baseBranch to new_branch

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Does GitLab allow two MRs targeting different branches but from the same MR branch?

I'm not sure we can easily/automatically change the base branch, because then in the case that someone modifies a Renovate MR intentionally, Renovate will then modify it right back.

Instead, I think that if someone ticks the rebase/retry checkbox, that should tell us it's ok to make the base branch change.

**Reply by alreadyhaveanaccount4** [reply]
As far as I've read, this is a requested Feature but not yet supported by GitLab.

We used the old_branch to make some test-runs and not want to switch to the new_branch. What would be the appropriate way to do this? As mentioned, we checked the retry/rebase checkbox. We can't delete MRs, only Branches.

**Reply by rarkins** [reply]
Are you seeing the "PR has been edited" log messages here? https://github.com/renovatebot/renovate/blob/e8ecac83731a8502c8da28155cdd0c1f330c4271/lib/workers/repository/update/branch/index.ts#L201-L215

**Comment by alreadyhaveanaccount4** [comment, marked as answer]
Yes, I do. We have Added some manual changes to it.

**Reply by rarkins** [reply]
Please elaborate on what you mean

**Reply by alreadyhaveanaccount4** [reply]
The MR opened by renovate had some linting-mismatches, which we then resolved manually by adding a commit to the open MR.

**Reply by rarkins** [reply]
In this case if you have made modifications *and* need the base branch changed, I don't think we'll ever support resolving that automatically. You will need to resolve the target change manually.

**Reply by alreadyhaveanaccount4** [reply]
What exactly do you mean with `You will need to resolve the target change manually.` ? Would it be enough if we close the MR?

**Reply by rarkins** [reply]
If you ever want an MR recreated, you need to close the MR *and* rename it.

But I think in this case you'd lose your changes, as Renovate would subsequently take over the branch and push to it.

Best if you take a copy of the branch (i.e. branch off it) before doing this so that you can then merge the changes back into the branch once the clean MR is created

**Reply by alreadyhaveanaccount4** [reply]
I'll try doing this. Thanks for your help.
